compiler/rustc_interface/src/passes.rs

```diff
@@ -632,6 +632,16 @@ fn write_out_deps(
             }
         }
 
+        let file_depinfo = sess.parse_sess.file_depinfo.borrow();
+        if !file_depinfo.is_empty() {
+            let mut paths: Vec<_> = file_depinfo.iter().map(|path| escape_dep_env(*path)).collect();
+            paths.sort_unstable();
+            writeln!(file)?;
+            for path in paths {
+                writeln!(file, "# file-dep:{}", path)?;
```

Comment:
`# file-dep` comments are not necessary, makefile syntax natively supports depending on files.

See the logic processing `files` above the env depinfo logic.
Files from `parse_sess.file_depinfo` need to be added there instead.